### PR TITLE
Backport Scintilla patch correctly disposing scrollbars

### PR DESCRIPTION
**Pull request by techee**
The unparent method should be called in dispose(), not in finalize().
This patch fixes that which removes some ugly warnings from OS X command
line.

More here:

http://sourceforge.net/p/scintilla/bugs/1777/

Would be nice to have this for 1.26 to get rid of the warnings.

### CODE REVIEW

**Comment by techee** [comment]
@b4n Added the NULL checks. If you find it too risky for 1.26, I can also just cherry-pick the patch and use it only for the OS X binaries.

**Comment by b4n** [comment]
Well, I find it to look quite OK, but I can't reproduce any of the issues you and Neil have been seeing, so I'm a bit lost in what to trust (I believe you're both right, but how I don't understand :).

So for the moment I'll hold this back and let you cherry-pick it for OSX, and if Neil answers me quickly I'll see.

@codebrainz BTW, could you confirm the issue @techee's been having on OSX?

**Comment by techee** [comment]
@b4n Unless @codebrainz compiled the latest Geany from git, he won't see it - this wasn't present in the previous Geany release (you added the leak-fixing patch on July 5, I guess previous Geany shipped with an older version of Scintilla) and I didn't see these warnings with the previous official OS X binary.

No problem for me to just cherry-pick the patch.

**Comment by codebrainz** [comment]
Also, I'm not sure the exact issue, the link in PR description brings to a page that says:

> The sourceforge.net website is temporarily in static offline mode. Only a very limited set of project pages are available until the main website returns to service.

**Comment by b4n** [comment]
> Also, I'm not sure the exact issue, the link in PR description brings to a page that says: […]

Here's a copy of the description of the symptoms from the initial report:

> On OS X with GTK backend I get the following warning on destruction:
>
> (geany:70109): GLib-GObject-CRITICAL **: g_object_ref: assertion 'object->ref_count > 0' failed
>
> twice for every closed editor in Geany.

**Comment by techee** [comment]
@codebrainz You should be able to reproduce it with the binary here

http://download.geany.org/snapshots/geany-1.26_osx.dmg

and see the warnings when running Geany from the terminal using

/Applications/Geany.app/Contents/MacOS/geany

File: scintilla/gtk/ScintillaGTK.cxx [hunk 1]
```diff
@@ -3114,6 +3135,7 @@ void ScintillaGTK::ClassInit(OBJECT_CLASS* object_class, GtkWidgetClass *widget_
 	// in Initialise() may require coordinate translation?)
 
 	object_class->finalize = Destroy;
+	object_class->dispose = Dispose;
```
Comment (b4n) [inline scintilla/gtk/ScintillaGTK.cxx]: doesn't matter, but order is inverted from upstream Scintilla

**Comment by b4n** [comment]
Merged, as @codebrainz confirmed same issue on OSX, and Neil confirmed the NULL check fixed is weird issue.

**Pull request merged**